DM-15871: Move doImport out of daf_butler.

TallJimbo opened the pull request.

TallJimbo committed:
Move doImport out of daf_butler.

timj commented on python/lsst/daf/butler/butler.py:
```diff
@@ -27,7 +27,8 @@
 import logging
 
-from .core.utils import doImport, transactional
+from lsst.utils import doImport
```
You probably should move the lsst.utils imports before the butler imports.

TallJimbo replied:
It is already, isn't it?

timj replied:
😄 I think there might be one that isn't later on... Sorry.

timj reviewed (approved):
Only comment is to fix the order of all the imports so that lsst.utils is above the butler internal imports.

TallJimbo committed:
Clean up import order and conventions.

Note that we use relative and complete imports within the core module,
but absolute and minimal imports in the "implementation" subpackages.

TallJimbo commented:
@timj I've pushed a broader cleanup to the import order and style (since I found some pre-existing things).  Please make sure you don't hate it.

timj reviewed (approved):
Looks good to me.
